src/panels/lovelace/common/evaluate-filter.ts

```diff
@@ -20,7 +20,26 @@ export const evaluateFilter = (stateObj: HassEntity, filter: any): boolean => {
       return state > value;
     case "!=":
       return state !== value;
+    case "in":
+      if (Array.isArray(state) || typeof state === typeof "string") {
+        return state.includes(value);
+      }
+      return false;
+    case "not in":
+      if (Array.isArray(state) || typeof state === typeof "string") {
+        return !state.includes(value);
+      }
+      return false;
     case "regex": {
+      if (
+        typeof state !== typeof "string" &&
+        typeof state !== typeof 1 &&
+        typeof state !== typeof true &&
+        state !== null &&
+        typeof state !== typeof undefined
```

Comment:
Wouldn't this be enough?
```suggestion
   typeof state === "object"
```

Reply:
For this instance it would be enough since stringify will take any type of object, even null.

Reply:
@bramkragten Actually I changed it to ```state !== null && typeof state === "object"```. You don't want null in stringify because it will evaluate to ```"null"``` and could create false positives.